File: jsimple-oauth/src/main/java/jsimple/oauth/exceptions/OAuthParametersMissingException.java
/*
 * Portions copyright (c) 2012-2014 Microsoft Mobile.  All Rights Reserved.
 *
 * This file is based on or incorporates material from the Scribe Java OAuth
 * library https://github.com/fernandezpablo85/scribe-java (collectively, "Third
 * Party Code"). Microsoft Mobile is not the original author of the Third Party
 * Code.
 *
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

package jsimple.oauth.exceptions;

import jsimple.oauth.model.OAuthRequest;

/**
 * Specialized exception that represents a missing OAuth parameter.
 *
 * @author Pablo Fernandez
 */
public class OAuthParametersMissingException extends OAuthException {
    private static final long serialVersionUID = 1745308760111976671L;

    /**
     * Default constructor.
     *
     * @param request OAuthRequest that caused the error
     */
    public OAuthParametersMissingException(OAuthRequest request) {
        super("Could not find oauth parameters in request: " + request +
                ". OAuth parameters must be specified with the addOAuthParameter() method");
    }
}

File: jsimple-oauth/c#/jsimple/oauth/exceptions/OAuthParametersMissingException.cs
namespace jsimple.oauth.exceptions
{

	using OAuthRequest = jsimple.oauth.model.OAuthRequest;

	/// <summary>
	/// Specialized exception that represents a missing OAuth parameter.
	/// 
	/// @author Pablo Fernandez
	/// </summary>
	public class OAuthParametersMissingException : OAuthException
	{
		private const long serialVersionUID = 1745308760111976671L;

		/// <summary>
		/// Default constructor.
		/// </summary>
		/// <param name="request"> OAuthRequest that caused the error </param>
		public OAuthParametersMissingException(OAuthRequest request) : base("Could not find oauth parameters in request: " + request + ". OAuth parameters must be specified with the addOAuthParameter() method")
		{
		}
	}

}